pkg/ingester/ingester_v2.go

```diff
@@ -48,12 +48,13 @@ type Shipper interface {
 }
 
 type userTSDB struct {
-	*tsdb.DB
-	userID         string
-	refCache       *cortex_tsdb.RefCache
-	activeSeries   *ActiveSeries
-	seriesInMetric *metricCounter
-	limiter        *Limiter
+	db               *tsdb.DB
+	userID           string
+	refCache         *cortex_tsdb.RefCache
+	activeSeries     *ActiveSeries
+	seriesInMetric   *metricCounter
+	limiter          *Limiter
+	compactBlocksMtx sync.Mutex
```

Comment:
Typically mutex is on top of the block of variables that it protects. This mutex is used differently though. I would suggest adding a comment about it.

Reply:
Actually... I don't understand *why* we even need this, given that compaction can *only* be done from our own loop, and we never compact the same block concurrently.